front/src/app/shared/task-list-manager/task-list-manager.service.ts

```diff
@@ -33,14 +33,22 @@ export class TaskListManagerService {
     this.apiService.getLists()
       .pipe(takeUntil(this.destroy$))
       .subscribe(data => {
-        this.allTaskListsSubject$.next(data)
+        this.allTaskListsSubject$.next(data);
+        this.activeListControlIndexSubject$.next(0);
       });
   }
 
   deleteTaskListById(id: string): void { 
     this.apiService.deleteTaskList(id)
       .subscribe(() => {
-        this.router.navigate([]);
+        this.listId$
+          .pipe(takeUntil(this.destroy$))
+          .subscribe(listId => {
+            if (listId !== id) {
+              return

```

Comment:
Having `subscribe` inside of another `subscribe` considered to be a bad practice.

Try using `withLatestFrom` rxjs operator in the main observable pipe to get `listId`